crates/bounded-executor/src/concurrent_stream.rs

```diff
@@ -0,0 +1,151 @@
+use crate::BoundedExecutor;
+use futures::{
+    stream::{self, FusedStream},
+    Future, FutureExt, Stream, StreamExt,
+};
+
+pub fn concurrent_map<St, Fut, F>(
+    stream: St,
+    executor: BoundedExecutor,
```

Comment:
I don't feel we need to pass the bounded executor in here since the flat_map_unordered already accepts a limit?

I'd imagine something like this
```
flat_map_unordered(8, move |item| {
  stream::once(tokio::spawn(mapper(item))
})
```

Reply:
It does, but the limit is bound to that operation right. One thing I found is we clone the BoundedExecutor and share between epoch manager, pipeline, etc, so we globally limit the maximum tasks across all the places it is shared. I found it to be a good design choice to limit the number of outstanding verification tasks in the whole of consensus. That's why I passed in the bounded executor.

Reply:
i see, I don't have strong opinion on local vs global limit but I think even we use bounded executor we should set limit for the first flat_map_unordered, otherwise it can create a lot of futures awaiting to get permit from the executor